.travis.yml

```diff
@@ -6,60 +6,52 @@ services:
+    - COMPILER=gnu BUILD=Release
+    - COMPILER=gnu BUILD=Release SHARED_LIBS=off
+
+    - COMPILER=clang BUILD=Debug
+    - COMPILER=clang BUILD=Debug SHARED_LIBS=off
+    - COMPILER=clang BUILD=Release
+    - COMPILER=clang BUILD=Release SHARED_LIBS=off
+
+    - COMPILER=gnu BUILD=Release SONARQUBE=ON
 
 script:
+  - export DISTRO=alpine
   - rm -rf tpl
   - cp -vr docker ${HOME}/docker
-  - if [[ ${CC} != gcc ]]; then TAG="-${CC}"; fi
-  - sed -i "1s/gentoo/${DISTRO}${TAG}/" ${HOME}/docker/Dockerfile.quinoa-build-travis
   - cd ../../
   - mv -v ${TRAVIS_REPO_SLUG} $HOME/docker/quinoa
   - cp -r $HOME/.sonar ${HOME}/docker/sonar
-  - docker build --build-arg CXXFLAGS=${WERROR:+-Werror}
-                --build-arg COVERAGE=${COVERAGE}
-                --build-arg CC=${CC} --build-arg CXX=${CXX}
-                --build-arg RUNNER_ARGS="${RUNNER_ARGS}"
-                --build-arg TPL_DIR="${TPL_DIR}"
-                --build-arg SONARQUBE=${SONARQUBE} --build-arg SONARQUBE_TOKEN=${SONARQUBE_TOKEN} --build-arg SONARQUBE_GITHUB_TOKEN=${SONARQUBE_GITHUB_TOKEN}
-                --build-arg TRAVIS_BRANCH=${TRAVIS_BRANCH} --build-arg TRAVIS_JOB_NUMBER=${TRAVIS_JOB_NUMBER}
-                --build-arg TRAVIS_PULL_REQUEST=${TRAVIS_PULL_REQUEST} --build-arg TRAVIS_JOB_ID=${TRAVIS_JOB_ID}
-                --build-arg TRAVIS_TAG=${TRAVIS_TAG} --build-arg TRAVIS_REPO_SLUG=${TRAVIS_REPO_SLUG}
-                --build-arg TRAVIS_COMMIT=${TRAVIS_COMMIT}
-                -t ${TRAVIS_REPO_SLUG}:${DISTRO}${TAG} -f ${HOME}/docker/Dockerfile.quinoa-build-travis ${HOME}/docker/ &&
-    CON=$(docker run -d ${TRAVIS_REPO_SLUG}:${DISTRO}${TAG} /bin/bash) &&
+  - docker build --build-arg COMPILER=${COMPILER}
+                 --build-arg BUILD=${BUILD}
+                 --build-arg SHARED_LIBS="${SHARED_LIBS}"
+                 --build-arg SONARQUBE=${SONARQUBE}
+                 --build-arg SONARQUBE_TOKEN=${SONARQUBE_TOKEN}
+                 --build-arg SONARQUBE_GITHUB_TOKEN=${SONARQUBE_GITHUB_TOKEN}
+                 --build-arg TRAVIS_BRANCH=${TRAVIS_BRANCH}
+                 --build-arg TRAVIS_JOB_NUMBER=${TRAVIS_JOB_NUMBER}
+                 --build-arg TRAVIS_PULL_REQUEST=${TRAVIS_PULL_REQUEST}
+                 --build-arg TRAVIS_JOB_ID=${TRAVIS_JOB_ID}
+                 --build-arg TRAVIS_REPO_SLUG=${TRAVIS_REPO_SLUG}
+                 --build-arg TRAVIS_COMMIT=${TRAVIS_COMMIT}
+                 -t ${TRAVIS_REPO_SLUG}:${DISTRO}
+                 -f ${HOME}/docker/Dockerfile.quinoa-build-travis
+                 ${HOME}/docker/ &&
+    CON=$(docker run -d ${TRAVIS_REPO_SLUG}:${DISTRO} /bin/bash) &&
     docker cp ${CON}:/home/quinoa/.sonar ${HOME}/
 
-after_success:
-  - if [[ ${DOCKERHUB} = true && ${DOCKER_USERNAME} && ${DOCKER_PASSWORD} && ${TRAVIS_PULL_REQUEST} == false && ${TRAVIS_BRANCH} == master ]]; then
-      docker login -u="$DOCKER_USERNAME" -p="$DOCKER_PASSWORD";
-      docker push "${TRAVIS_REPO_SLUG}:${DISTRO}${TAG}";
-    fi
```

Comment:
Didn't you want to put that back?

Reply:
Yeah! I think so.

Reply:
Can you just push it?

Reply:
Yeah. I misunderstood your question. I will out this back and merge. Thanks for taking a look.